### ISSUE TITLE
Support BigQuery Materialized Views

### ISSUE THREAD

**Issue by ran-eh** [dbt-labs/dbt-bigquery#18]
### Describe the feature
Support https://cloud.google.com/bigquery/docs/materialized-views-intro

### Describe alternatives you've considered
1. add BQ specific configuration:
```
{{ config(
    materialized='view',
    materialized-view='true' # default 'false
)}}
```
2. create new materialization, initially for BQ only, over time perhaps implement for additional databases
```
{{ config(
    materialized='materialized-view',
)}}
```

### Additional context
Some powerful stuff here https://cloud.google.com/bigquery/docs/materialized-views-intro

### Who will this benefit?
BQ users

**Comment by ian-wesley** [comment]
This would be very helpful for my current use case.  Any update on when this is coming?

**Comment by var-bolt** [comment]
+1

**Comment by NoamMosko** [comment]
+1 :)

**Comment by sboyaci-nakd** [comment]
+1

**Comment by THEdavehogue** [comment]
+1

**Comment by enzorooo** [comment]
+1

**Comment by pmunari** [comment]
+1

**Comment by araa47** [comment]
+1

**Comment by ytaniguchi8** [comment]
+1 :)

**Comment by Bruno1298** [comment]
+1

**Comment by borislitvak** [comment]
+1

**Comment by buckybush** [comment]
+1

**Comment by gscs-data** [comment]
+2

**Comment by octavio-melo** [comment]
+1

**Comment by bilus** [comment]
+1

**Comment by dinigo** [comment]
+1

**Comment by willbryant** [comment]
Will you guy knock it off? The way to vote on things is to use the 👍 reaction on the description at the top, not to post comments.

**Comment by dinigo** [comment]
@willbryant :  Upvotes won't prevent the issue from going "stale" and being automatically archived [by the bot](https://github.com/marketplace/stale).

Also upvotes wont bump the issue, so developers will not have a sense of priority.

But you can customise your subscription to prevent "alarm fatigue" (spam)
![image](https://user-images.githubusercontent.com/3038962/185103055-353a86b4-ba3e-4338-baab-061bb8de3150.png)

**Comment by marcoruggine** [comment]
+1, this is essential

**Comment by SPTKL** [comment]
+1

**Comment by ee-jayson** [comment]
+ This is so needed!

**Comment by minhajpasha** [comment]
+1

**Comment by dreinon** [comment]
+1

**Comment by yuganghu** [comment]
+1

**Comment by artemboiko1** [comment]
+1

**Comment by techadmin-getsales** [comment]
+1

**Comment by geerdink** [comment]
+1

**Comment by daniel84101** [comment]
+1

**Comment by dinigo** [comment]
This was implemented in dbt v1.7  https://github.com/dbt-labs/dbt-core/pull/8449

Here an article from their blog
https://docs.getdbt.com/blog/announcing-materialized-views
